app/controllers/users/phone_confirmation_controller.rb

```diff
@@ -40,11 +40,13 @@ def assign_phone
                           phone_confirmed_at: Time.current)
     end
 
-    def after_confirmation_path
+    def after_phone_confirmation
       if @updating_existing_number
-        profile_path
+        create_user_event(:phone_changed)
+        redirect_to profile_path
       else
-        after_sign_in_path_for(current_user)
+        create_user_event(:phone_confirmed)
+        redirect_to after_sign_in_path_for(current_user)
```

Comment:
What do you think about keeping event-related actions together? I would leave this method as it was, and place the `create_user_event(:phone_changed)` in `assign_phone` next to the analytics event. And the `:phone_confirmed` event can go in `PhoneConfirmationFlow#confirm`. That way, you are extending the code without modifying it. You can call the event creation in existing places without needing to make these modifications in several places.